distribution/client_test.go

```diff
@@ -26,16 +29,13 @@ var (
 
 func TestClientPullModel(t *testing.T) {
 	// Set up test registry
-	registryContainer, err := tc.Run(context.Background(), "registry:2.8.3")
+	server := httptest.NewServer(registry.New())
```

Comment:
why do you prefer to use a "real" server vs TC in this case? just to learn 🙏 
One benefit of using TC here is that we have random port

Reply:
TC is obviously a good solution as well. I went with the httptest server this time b/c
* it eliminate the only dependency our tests have on docker - so we could theoretically run unit tests on github hosted mac and windows runners (which last I checked don't have docker install by default)
* tests run faster if we don't have to wait for the container to start/stop
* major reduction in the number of transitive dependencies (including some packages in docker and containerd that could maybe possibly lead to version conflict issue in the future?)

That said, none of the above are insurmountable challenges and if we felt the need for TC in the future we can use it. Just in this particular case, there is another solution (also with random ports) that felt equally convenient to me.